src/chromium/cargo.md

```diff
@@ -1,56 +1,53 @@
 
-```bob
-High        ^
-            |      x cargo
-            |
-Development |                     x "cargo --offline"
-speed       |
-            |                                  x  "gn/ninja"
-            |                                     "rust_executable(...)" 
-Low         +---------------------------------------------------->
-             Low               Determinism                 High
-```
-
-`cargo` works great for pure-Rust tools, but isn't optimized for large multi-
-language projects like Chromium. Chromium uses `gn` and `ninja`.
-
-When writing a tool in Rust, your choices are:
+When writing code in Rust, your choices are:
 
 * Use `gn` and `ninja` (using the `rust_executable` template we'll meet
   later)
 * Use `cargo`, but [restrict yourself to Chromium's audited toolchain and crates][0]
 * Use `cargo`, trusting a [toolchain][1] and [crates downloaded from the internet][2]
 
-Your organization's policy, and/or common sense, may prohibit you from doing
-these things.
-
 From here on we'll be focusing on `gn` and `ninja`.
 
 ## Mini exercise
 
-Discuss in small groups the policies within your own team and organization,
-and come to a group agreement about what's an acceptable level of risk.
+Discuss in small groups the policies within your own team and
+organization, and come to a group agreement about your assessment
+of risk associated with the choices above.
 
 <details>
 
-Talk about the cultural differences between the `cargo` world and the Chromium
-world: for instance, that code reuse is cheap in `cargo` and that it's
-encouraged to use (and create) lots of small single-purpose crates, which is
-difficult in Chromium (for good reasons). The net effect is that `cargo`
-ecosystem development feels more agile.
-
-Explain that it might seem strange to write tools in Rust, but this is
-increasingly popular across the industry --- Rust tools are quicker and work
-more reliably.
-
 Assuming folks taking the course are physically together, ask them to discuss
 in small groups of 3-4 people. Then, ask each table whether they've come
 to a consensus on the level of risk.
 
 Later in the course, we'll be running an actual `cargo`-based tool, `gnrt`.
+(`gnrt` uses Chromium's copy of `cargo` and `rustc`, but depends on third-party
+libraries downloaded from the internet, although `run_gnrt.py` asks `cargo` that
+only `--locked` contents are allowed via `Cargo.lock`.)
+
+Depending on the discussion, consider also mentioning the points below:
+
+* Talk about the cultural differences between the `cargo` world and the Chromium
+  world. Point out that one reason for adopting Rust in Chromium is to tap into
+  the rich, vibrant ecosystem of third-party Rust crates.
+* Talk about different risk profiles of code of the Chromium browser, Chromium
+  tests and tools, and other projects.  Point out that command like tools
+  authored in Rust are increasingly popular across the industry --- Rust tools
```

Comment:
I'd like to retain mention of "surprise" here. Many people are apparently, _surprised_ by the fact that Rust is used as a quick and easy language for tools which would previously have been written in python. People think of Rust as a heavyweight systems language; I think many audiences at this point will share surprise that it can also be used for quick + hacky tools, and talking about that surprise shows empathy with the audience.